_gsocproposals/2020/proposal_ROOTIO.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Pre-conditioners applied to ROOT compression algorithms
+layout: gsoc_proposal
+project: ROOT
+year: 2020
+organization:
+  - UNL
+  - MIR
+---
+
+## Description
+
+Pre-conditioning is an algorithm that rearranges typed, binary data to improve compression. More information can be found here: [1],[2]
+
+The idea of the project is that different filter or transformation schemes that could be applied to ROOT data, which are stored in "baskets". 
+
+In ROOT, the serialization of variable-sized data (containing C-style arrays) produces two internal arrays: one contains the branch data for each event while the other contains the byte offset of each event in the branch data. LZ4 compression algorithm achieves its performance by looking for byte-aligned patterns (as opposed to ZLIB compresison algorithm, which works on individual bits) and lacks the Huffman encoding pass, this results in the offset array sequence being effectively incompressible using LZ4.
```

Comment:
"the branch data for each event" -> maybe "the branch data for several events"? Otherwise why do you need event offsets?

Reply:
Maybe add links or references to LZ4 and ZLIB

Reply:
I would also add a direct link on keywords like "Bitshuffle" in addition to the reference to the bibliography at the end. It makes the reader happier...

Reply:
Thanks a lot, it is fixed! I really appreciate feedback!